### PR TITLE
handle damage to the view

### PR DESCRIPTION
**Pull request by watasuke102**
## Context

See #122

## Summary

`zn_view_damage()` fetches damage by `wlr_surface_get_effective_damage()`, and apply it.

## How to check behavior

start zen with some client, check that the view is rendered correctly.

### CODE REVIEW

**Review by Aki-7 — DISMISSED** [review]

File: zen/scene/view.c [hunk 1]
```diff
@@ -20,27 +20,54 @@ zn_view_move(struct zn_view *self, double x, double y)
...
+  for (int i = 0; i < rect_count; ++i) {
+    damage_box = (struct wlr_fbox){
+        .x = self->x + rects[i].x1,
+        .y = self->y + rects[i].y1,
+        .width = rects[i].x2 - rects[i].x1,
+        .height = rects[i].y2 - rects[i].y1,
+    };
+    zn_view_add_damage_fbox(self, &damage_box);
+  }
+
```
[the remark this answers is not in the document]
Comment (watasuke102) [reply zen/scene/view.c:59]: In my understanding, the result of `wlr_surface_get_effective_damage` includes subsurfaces' damage. `wlr_surface` has subsurface's damage as `external_damage` (from reading header's comment). And, `external_damage` is used at `wlr_surface_get_effective_damage`.

So, I'll add comment only about popup.

[the remark this answers is not in the document]
Comment (Aki-7) [reply zen/scene/view.c:59]: > In my understanding, the result of wlr_surface_get_effective_damage includes subsurfaces' damage.

It's really complicated, but no, it's not. The "external damage" includes the damage when subsurfaces are moved, but does not include the damage of the subsurfaces' buffers.
In other words, when subsurfaces are not moved, and it's surface content has changed, the damaged area of the surface content will not be included in the "external damage". We have to iterate the subsurfaces and add the damage of "synced" subsurfaces.

[the remark this answers is not in the document]
Comment (Aki-7) [reply zen/scene/view.c:59]: so,

```suggestion

  // FIXME: add damages of "synced" subsurfaces and popups

```

is better, actually.

**Review by Aki-7 — APPROVED** [review]
LGTM!

**Pull request merged**